akka-stream-tests/src/test/scala/akka/stream/io/FileSourceSpec.scala

```diff
@@ -242,6 +244,13 @@ class FileSourceSpec extends StreamSpec(UnboundedMailboxConfig) {
       }
     }
 
+    "complete materialized future with a failure if upstream fails" in {
+      val matVal: Future[IOResult] =
+        FileIO.fromPath(manyLines, chunkSize = 4).map(_ => throw new RuntimeException).to(Sink.ignore).run()
+
+      matVal.failed.futureValue shouldBe a[IOOperationIncompleteException]
```

Comment:
Not quite sure about this, leaking the downstream exception might be a bad idea.

Reply:
I guess the question is, why does a Source return a `Future` as a materialized value in the first place? At the source side, you might have read something but there's no guarantee anything was done with the data that was read. It seems that whatever we do, people might be misled about what was going on...

Reply:
Yeah, that's a good point, I guess we can't quite change that now but we can document more clearly document that it doesn't guarantee anyone actually saw those bytes...

Reply:
Added a note in the docs about that